### PR TITLE
Feature/ung 3136 poc loop

### PR DESCRIPTION
**Pull request by Hiroki6**
I did these things:

- use Cryptshare instread of TeamDrive
- when pocType is API, send poc into certify api in poc loop, when pocType is APP, do it in poc admin loop
  - remove dependency of PocStatus from InformationProvider.infoToCertifyAPI
- create shared cert and send it to admin in poc admin loop
  - remove dependency of PocStatus from createPoCSharedAuthCertificate.PoCCertCreator
  - change PocAdminAndStatus into PocAdminTriple to have Poc object

We don't need `client_cert_created` and `client_cert_provided` status in the PocStatus object anymore?

### CODE REVIEW

**Review by lalottacontinua — APPROVED** [review]
sorry, not fully finished. too many interruptions. If you want to, we might go over it together once more..

File: poc-api/src/main/scala/com/ubirch/PocConfig.scala [hunk 1]
```diff
@@ -23,7 +23,7 @@ trait PocConfig {
   val locationNeeded: Seq[String]
   val roleNeeded: Seq[String]
 
-  val teamDriveAdminEmails: Seq[String]
+  val cryptShareAdminEmails: Seq[String]
   val teamDriveStage: String
```
[the remark this answers is not in the document]
Comment (Hiroki6) [reply poc-api/src/main/scala/com/ubirch/PocConfig.scala:27]: yes, later when Tenant Admin part adjusts Cryptshare.

File: poc-api/src/main/scala/com/ubirch/services/poc/DeviceCreator.scala [hunk 2]
```diff
@@ -132,11 +139,12 @@ class DeviceCreatorImpl @Inject() (conf: Config, aESEncryption: AESEncryption)(i
     write[DeviceRequestBody](body)
   }
 
-  protected def decryptToken(tenant: Tenant, poc: Poc, status: PocStatus): Task[DecryptedData] = {
+  protected def decryptToken(tenant: Tenant): Task[DecryptedData] = {
     if (tenant.deviceCreationToken.isEmpty)
-      throwError(PocAndStatus(poc, status), "poc.deviceGroupId is missing; cannot create device")
-    aESEncryption
-      .decrypt(tenant.deviceCreationToken.get.value)(identity)
+      Task.raiseError(InternalBatchException("poc.deviceGroupId is missing; cannot create device"))
```
[the remark this answers is not in the document]
Comment (Hiroki6) [reply poc-api/src/main/scala/com/ubirch/services/poc/DeviceCreator.scala:144]: oh, it's wrong. will fix it. thanks!

File: poc-api/src/main/scala/com/ubirch/services/poc/DeviceCreator.scala [hunk 3]
```diff
@@ -95,8 +105,7 @@ class DeviceCreatorImpl @Inject() (conf: Config, aESEncryption: AESEncryption)(i
   @throws[PocCreationError]
```
[the remark this answers is not in the document]
Comment (Hiroki6) [reply poc-api/src/main/scala/com/ubirch/services/poc/DeviceCreator.scala:105]: here is correct. I didn't change the device creation part.

File: poc-api/src/main/scala/com/ubirch/services/poc/InformationProvider.scala [hunk 4]
```diff
@@ -98,76 +99,51 @@ formats: Formats)
...
+        .flatMap {
           _.code match {
             case Ok | Created =>
-              statusAndPW.status.copy(certifyApiProvided = true)
+              Future.unit
             case code =>
-              throwError(
-                PocAndStatus(poc, statusAndPW.status),
-                s"failure when providing device info to certifyAPI, errorCode: $code")
+              Future.failed(
```
[the remark this answers is not in the document]
Comment (Hiroki6) [reply poc-api/src/main/scala/com/ubirch/services/poc/InformationProvider.scala:129]: ah, no. will revise other parts as well, thanks

File: poc-api/src/main/scala/com/ubirch/services/poc/PocAdminCreator.scala [hunk 5]
```diff
@@ -119,85 +121,68 @@ class PocAdminCreatorImpl @Inject() (
...
+          pocAdminTriple.updatePoc(_.copy(sharedAuthCertId = Some(sharedAuthResponse.certUuid)))
+            .updateStatus(_.copy(clientCertCreated = Some(true), clientCertProvided = Some(true)))
+        }.onErrorHandle {
+          case ex =>
+            PocAdminCreator.throwError(pocAdminTriple, ex.getMessage)
         }
-    } else Task(aAs)
+      } else {
+        // The other PocAdmin belonging to the same PoC already got the certificate
+        Task(pocAdminTriple.copy(status = pocAdminTriple.status.copy(clientCertCreated = Some(true))))
```
[the remark this answers is not in the document]
Comment (Hiroki6) [reply poc-api/src/main/scala/com/ubirch/services/poc/PocAdminCreator.scala]: 🤔

File: poc-api/src/main/scala/com/ubirch/services/poc/PocAdminCreator.scala [hunk 6]
```diff
@@ -119,85 +121,68 @@ class PocAdminCreatorImpl @Inject() (
...
-              logger)
+  private def createClientCertAndTransfer(pocAdminTriple: PocAdminTriple): Task[PocAdminTriple] = {
+    if (pocAdminTriple.poc.typeIsApp && pocAdminTriple.status.clientCertCreated.contains(false)) {
+      if (pocAdminTriple.poc.sharedAuthCertId.isEmpty) {
+        val emails = pocConfig.cryptShareAdminEmails.toList.::(pocAdminTriple.admin.email)
+        PoCCertCreator.createPoCSharedAuthCertificate(pocAdminTriple.poc, emails)(
+          certHandler,
+          cryptShareClient).map { sharedAuthResponse =>
+          pocAdminTriple.updatePoc(_.copy(sharedAuthCertId = Some(sharedAuthResponse.certUuid)))
+            .updateStatus(_.copy(clientCertCreated = Some(true), clientCertProvided = Some(true)))
```
[the remark this answers is not in the document]
Comment (Hiroki6) [reply poc-api/src/main/scala/com/ubirch/services/poc/PocAdminCreator.scala:159]: You're right, I will delete one

**Pull request merged**